spec/lib/msf/ui/console/command_dispatcher/core_spec.rb

```diff
@@ -397,4 +397,295 @@ def set_tabs_test(option)
+        TABLE
+      end
+    end
+
+    context "searches with sessions with different checkin values" do
+      let(:sessions) do 
+        {
+          1 => instance_double(::Msf::Sessions::Meterpreter_x64_Win, last_checkin: Time.parse('Dec 18, 2022 12:33:40.000000000 GMT'), type: 'meterpreter', sid:1, sname: "session1", info: "info", session_host: "127.0.0.1", tunnel_to_s: "tunnel"),
+          2 => instance_double(::Msf::Sessions::Meterpreter_x64_Win, last_checkin: (Time.parse('Dec 18, 2022 12:33:40.000000000 GMT') - 90), type: 'meterpreter', sid:2, sname: "session2", info: "info", session_host: "127.0.0.1", tunnel_to_s: "tunnel"),
+          3 => instance_double(::Msf::Sessions::Meterpreter_x64_Win, last_checkin: (Time.parse('Dec 18, 2022 12:33:40.000000000 GMT') - 20000), type: 'meterpreter', sid:3, sname: "session3", info: "info", session_host: "127.0.0.1", tunnel_to_s: "tunnel")
+        }
+      end
+
+      it "filters by last checkin using fractions of a second" do
+        core.cmd_sessions("--search", "last_checkin:after:1m40.5s")
+        expect(@output.join("\n")).to match_table <<~TABLE
+          Active sessions
+          ===============
+
+            Id  Name      Type         Information  Connection
+            --  ----      ----         -----------  ----------
+            1   session1  meterpreter  info         tunnel (127.0.0.1)
+            2   session2  meterpreter  info         tunnel (127.0.0.1)
+        TABLE
+      end
+  
+      it "filters by last checkin using fractions of a minute" do
+        core.cmd_sessions("--search", "last_checkin:before:0.5m1s")
+        expect(@output.join("\n")).to match_table <<~TABLE
+          Active sessions
+          ===============
+
+            Id  Name      Type         Information  Connection
+            --  ----      ----         -----------  ----------
+            2   session2  meterpreter  info         tunnel (127.0.0.1)
+            3   session3  meterpreter  info         tunnel (127.0.0.1)
+        TABLE
+      end
+
+      it "warns the user when searching for an unknown checkin parameter" do
+        core.cmd_sessions("--search", "last_checkin:something:10s")
+        expect(@combined_output.join("\n")).to match_table <<~TABLE
+          Please specify before or after for checkin query. Ex: last_checkin:before:1m30s. Given: something
+        TABLE
+      end
+
+      it "filters by last checkin using before and after terms" do
+        core.cmd_sessions("--search", "last_checkin:after:200s last_checkin:before:30s")
+        expect(@output.join("\n")).to match_table <<~TABLE
+          Active sessions
+          ===============
+
+            Id  Name      Type         Information  Connection
+            --  ----      ----         -----------  ----------
+            2   session2  meterpreter  info         tunnel (127.0.0.1)
+        TABLE
+      end
+
+      it "returns no matches if after and before are swapped" do
```

Comment:
Maybe something like 
```suggestion
      it "When the user specifies a before time that is larger than the after time" do
```
would make it more clear what the intention here is and not include what we expect the result to be, only include the action we're taking and confirm our expectations later

Reply:
`When` is generally reserved for `context` blocks - https://www.betterspecs.org/#contexts